### ISSUE TITLE
Waterfox Current initially can not use (but can downgrade, or import from) a profile that was used by a superior build of Waterfox Current

### ISSUE THREAD

**Issue by grahamperrin** [MrAlex94/Waterfox#1241]
From <https://old.reddit.com/comments/dnmecu/-/f5fbxyk/>: 

> As far as I know: 
> 
> - for as long as Waterfox Current will be based on Firefox ESR 68, it should be possible for an inferior version e.g. `2019.10 (20191022131008)` to use a profile that was touched by a superior version

**Logically** – given (a) the Waterfox Current user base; (b) soft launch scenarios; and so on – it **should** be possible. Desirable, sooner or later. 

Actually it's not (or not yet) possible. 

A fluffy example: 

- Waterfox Current `2019.10 (20191015121210)` can not use a profile that was used by Waterfox Current `2019.10 (20191023104314)`. 

– "fluffy" because I can think of no reason for _anyone_ to prefer `20191015121210`. 

Inherent protection; the Mozilla code base. A sane base, but it's likely to be perceived as troublesome in a future Waterfox scenario, so it's worth raising this issue (meta, tracking: `#538`) – maybe to be closed `WONTFIX`, but let's have a record. 

<strike>With Waterfox Current alone, there is no easy workaround. </strike>False! Sorry. See https://github.com/MrAlex94/Waterfox/issues/1241#issuecomment-547165604 below. 

----

A workaround does exist, however it's as _ugly_ (potentially troublesome) as it is _easy_ so I'll make it a separate comment, to be voted down.

**Comment by grahamperrin** [comment]
Here's a screen recording of the ugly workaround. 

(I just realised, there's an alternative workaround that's sane, but I'll go ahead and share what's below as an example of an approach that's **bad**; likely to be troublesome.)

Attention to the command lines: 

- https://put.re/player/e9pwa2zV.mp4
- [Waterfox Classic as a bad workaround to profile protection in Waterfox Current.zip](https://github.com/MrAlex94/Waterfox/files/3780817/Waterfox.Classic.as.a.bad.workaround.to.profile.protection.in.Waterfox.Current.zip)

In a nutshell: 

- this workaround involves Waterfox Classic
- end users of Waterfox Classic <strike>are not </strike> may be likely to fall into this trap; <strike>Waterfox Current profile must be explicitly chosen </strike>#1246
- ugly because there may be unknown negative consequences of non-supported downgrade actions.

**Comment by grahamperrin** [comment]
### Sane workaround

At the command line: 

`--migration`

![image](https://user-images.githubusercontent.com/192271/67721610-e0b44300-f9ce-11e9-87ad-89ca1e0f9399.png)

**Comment by MrAlex94** [comment]
https://github.com/MrAlex94/Waterfox/commit/96d1f39987b16d86555302c3c7351c7825ca4412

So that was the change - I’ll see if I can reproduce that bug; I have no
idea why the builds would do that - there were no changes in the toolkit
code for this. Thanks!

**Comment by grahamperrin** [comment]
Reopening this as a documentation issue – #582 – I imagine a single-sentence, single-paragraph answer amongst FAQ with a link to a separate page about:  

- supported downgrades e.g. from Waterfox Current to Waterfox Current
- non-supported downgrade scenarios e.g. to Waterfox Classic
- import/migration. 

@MrAlex94 probably not a bug. 

Gut feeling: let's not attempt to override the protection.

**Comment by grahamperrin** [comment]
@MrAlex94 PS (no rush) if you can remove the quoted text from your comment, doing so will probably cease the link with bug tracking issue `#538`. I mean, I no longer see this as a bug `:-)`

**Comment by grahamperrin** [comment]
> Inherent protection; the Mozilla code base. A sane base, 

For readers who may be unfamiliar with this approach to protection: two screenshots borrowed from <https://www.reddit.com/comments/dnmecu/-/f5gl7r0/?context=1>.

![image](https://user-images.githubusercontent.com/192271/67723645-95049800-f9d4-11e9-8f64-f32928888d4f.png)

![image](https://user-images.githubusercontent.com/192271/67723694-a8affe80-f9d4-11e9-8687-1f52b4173bcb.png)

In this case (without showing any build ID): 

- [www/firefox](https://www.freshports.org/www/firefox/) `70.0_2,1` could not use a profile that was used by `70.0_3,1`. 

----

In as few words as possible: 

- both installations were of the same version of Firefox
- whilst one is recognisably 'older' than the other, the recognition is **not** time-based
- protection is a good thing.

**Comment by laniakea64** [comment]
@grahamperrin Did you try just starting Waterfox Current with `--allow-downgrade` command-line option?

**Comment by grahamperrin** [comment]
@laniakea64 even better! Thanks. Sorry I didn't think to check available options. 

https://developer.mozilla.org/docs/Mozilla/Command_Line_Options#User_Profile

Confirmed: the option is effective.

**Comment by grahamperrin** [comment]
Reopening solely for documentation purposes. Not a bug. 

https://github.com/MrAlex94/Waterfox/issues/1241#issuecomment-547168365 above is expanded to three bullet points.

**Comment by grahamperrin** [comment]
Reopening solely for documentation purposes. Not a bug. 

https://github.com/MrAlex94/Waterfox/issues/1241#issuecomment-547168365 above is expanded to three bullet points.
